### PR TITLE
tidied up generate.js

### PR DESCRIPTION
**Pull request by mbugrayildiz**
It runs exactly the same speed as before, the code is just prettier now.

### CODE REVIEW

**Review by tansionline — APPROVED** [review]
LGTM

File: generate.js [hunk 1]
```diff
@@ -121,23 +108,19 @@ const getData = async () => {
...
+      person.contributions = mentorships.filter((mentorship) => {
+        if (!person.github) return false
+
+        for (const contrib of mentorship.contributors) {
+          const personHasContributed = contrib.github_address == person.github
+          if (personHasContributed) return true
         }
+
+        return false
       })
```
Comment (ahmetcanaydemir) [inline generate.js:124]:
```suggestion
      person.contributions = mentorships.filter((mentorship) => {
            return person.github &&
            mentorship.contributors.some((contrib) => contrib.github_address === person.github)
      })
```

File: generate.js [hunk 2]
```diff
@@ -35,52 +35,41 @@ const mapper = (posts) => {
...
-  return posts.map((post) => {
-    post.project_adress = fixProtocol(post.project_adress)
-    const projectName = post.project_adress.split('/').pop()
-    post.slug = projectName
-    return post
+const clearMentorships = (mentorships) =>
+  mentorships.map((mentorship) => {
+    mentorship.project_adress = fixProtocol(mentorship.project_adress)
+    mentorship.slug = mentorship.project_adress.split('/').pop()
+    return mentorship
```
Comment (ahmetcanaydemir) [inline generate.js:71]:
```suggestion
    return {
        ...mentorship,
        project_adress : fixProtocol(mentorship.project_adress),
        slug: mentorship.project_adress.split('/').pop()
     }
```

File: generate.js [hunk 3]
```diff
@@ -35,52 +35,41 @@ const mapper = (posts) => {
...
+  people.map((person) => {
+    person.slug = slugger(person.name)
+    person.github = fixProtocol(person.github)
+    person.twitter_handle = fixProtocol(person.twitter_handle)
+    person.twitter_handle = clearTwitterAdr(person.twitter_handle)
+    person.linkedin = fixProtocol(person.linkedin)
+    person.avatar = generateAvatar(person)
+    person.displayInterests = person.interests
+    person.interests = person.interests.toLowerCase()
+    return person
```
Comment (ahmetcanaydemir) [inline generate.js:64]:
```suggestion
    return {
        ...person,
        slug: slugger(person.name),
        github: fixProtocol(person.github),
        twitter_handle: clearTwitterAdr(fixProtocol(person.twitter_handle)),
        linkedin: fixProtocol(person.linkedin),
        avatar: generateAvatar(person),
        displayInterests: person.interests,
        interests: person.interests.toLowerCase()
    }
```

File: generate.js [hunk 4]
```diff
@@ -35,52 +35,41 @@ const mapper = (posts) => {
...
-    avatar = `https://avatars.githubusercontent.com/${response[1]}`
-  } else {
-    avatar = `https://ui-avatars.com/api/?name=${name}&size=256`
-  }
-  return avatar
+  if (!github) return `https://ui-avatars.com/api/?name=${name}&size=256`
+
+  const regex = /github.com\/([\w\d-]+)(. +)?/
+  const response = github.match(regex)
+  if (!response) return ''
```
Comment (ahmetcanaydemir) [inline generate.js:42]:
```suggestion
  if (!github.match(/github.com\/([\w\d-]+)(. +)?/)) return ''
```

**Comment by cagataycali** [comment]
We will not gonna use in one week, that's the reason I'm closing this pr

**Pull request closed without merging**